### PR TITLE
update kernel configs, make kernel in integ test

### PR DESCRIPTION
**Pull request by ginglis13**
Signed-off-by: Gavin Inglis <giinglis@amazon.com>

*Issue #, if available:*

#658 

*Description of changes:*

in upgrading to firecracker v1.1, we noticed some configuration options
prompted for using when `make kernel`. These options are specific to
kernel 4.14. Additionally, this upgrade changed the build output from
firecracker's devtool to exclude the full kernel version, and installing would fail as it expects the path to include the patch number.

Some relevant links from the linux kernel driver db:

* https://cateee.net/lkddb/web-lkddb/KMEMCHECK.html
* https://cateee.net/lkddb/web-lkddb/CRYPTO_SALSA20_X86_64.html
* https://cateee.net/lkddb/web-lkddb/FRAME_POINTER_UNWINDER.html
* https://cateee.net/lkddb/web-lkddb/ORC_UNWINDER.html
* https://cateee.net/lkddb/web-lkddb/R3964.html

This PR also updates the integ test Docker image to use a freshly built
kernel such that FUSE is enabled.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tools/kernel-configs/microvm-kernel-x86_64-4.14.config [hunk 1]
```diff
@@ -1208,6 +1208,7 @@ CONFIG_VIRTIO_BLK=y
 # CONFIG_ENCLOSURE_SERVICES is not set
 # CONFIG_SRAM is not set
 # CONFIG_C2PORT is not set
+# CONFIG_R3964 is not set
```
Comment (Kern--) [inline tools/kernel-configs/microvm-kernel-x86_64-4.14.config:1211]: Are these x86 specific or do we also need these options in the ARM config?

File: tools/docker/Dockerfile.integ-test [hunk 2]
```diff
@@ -61,7 +61,8 @@ RUN --mount=type=bind,target=/src make -C /src \
   install \
   install-firecracker \
   install-runc \
-  install-default-vmlinux \
+  kernel \
+  install-kernel \
```
[the remark this answers is not in the document]
Comment (ginglis13) [reply tools/docker/Dockerfile.integ-test:65]: I see... write permission may have to do with not building the image as privileged. We can investigate having a pre-built kernel for this use case.

[the remark this answers is not in the document]
Comment (ginglis13) [reply tools/docker/Dockerfile.integ-test:65]: +1 to pre-built kernel image for avoiding increased build time. Here are some sample times running locally.

Without the firecracker image public.ecr.aws/firecracker/fcuvm:v35 not yet built: 

```
$ time make kernel
...
real    1m23.834s
```

With firecracker image already built:

```
$ time make kernel
...
real    0m47.606s
```

File: Makefile [hunk 3]
```diff
@@ -60,7 +60,7 @@ KERNEL_CONFIG=tools/kernel-configs/microvm-kernel-$(host_arch)-$(KERNEL_VERSION)
 # Copied from https://github.com/firecracker-microvm/firecracker/blob/v1.0.0/tools/devtool#L2015
 # This allows us to specify a kernel without the patch version, but still get the correct build path to reference the kernel binary
 KERNEL_FULL_VERSION=$(shell cat "$(KERNEL_CONFIG)" | grep -Po "^\# Linux\/$(kernel_config_pattern) (([0-9]+.){2}[0-9]+)" | cut -d ' ' -f 3)
```
Comment (swagatbora90) [inline Makefile:62]: I normally prefer to clean up unused defined variables. In this case looks like `KERNEL_FULL_VERSION` is no longer being used. I will let you decide if we need to keep this around for future use.

**Review by austinvazquez — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (ginglis13) [reply Makefile]: found the real reason - the regex for kernel version slightly changed in firecracker v1.1.0: https://github.com/firecracker-microvm/firecracker/blob/v1.1.0/tools/devtool#L2082 . updated the regex in our Makefile and reverted back to using `KERNEL_FULL_VERSION`

**Pull request merged**